homeassistant/components/hisense_aehw4a1/climate.py

```diff
@@ -0,0 +1,422 @@
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        if temp is not None:
+            _LOGGER.debug("Setting temp of %s to %s", self._unique_id, temp)
+            if self._preset_mode != PRESET_NONE:
+                await self.async_set_preset_mode(PRESET_NONE)
+            if self._temperature_unit == TEMP_CELSIUS:
+                await self._device.command(f"temp_{int(temp)}_C")
+            else:
+                await self._device.command(f"temp_{int(temp)}_F")
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set new fan mode."""
+        if self._on == "1":
+            if self._hvac_mode in (HVAC_MODE_COOL, HVAC_MODE_FAN_ONLY) and (
+                self._hvac_mode != HVAC_MODE_FAN_ONLY or fan_mode != FAN_AUTO
+            ):
+                _LOGGER.debug("Setting fan mode of %s to %s", self._unique_id, fan_mode)
+                await self._device.command(HA_FAN_MODES_TO_AC[fan_mode])
+
+    async def async_set_swing_mode(self, swing_mode):
+        """Set new target swing operation."""
+        if self._on == "1":
+            _LOGGER.debug("Setting swing mode of %s to %s", self._unique_id, swing_mode)
+            swing_act = self._swing_mode
+
+            if swing_mode == SWING_OFF and swing_act != SWING_OFF:
+                if swing_act in (SWING_HORIZONTAL, SWING_BOTH):
+                    await self._device.command("hor_dir")
+                if swing_act in (SWING_VERTICAL, SWING_BOTH):
+                    await self._device.command("vert_dir")
+
+            if swing_mode == SWING_BOTH and swing_act != SWING_BOTH:
+                if swing_act in (SWING_OFF, SWING_HORIZONTAL):
+                    await self._device.command("vert_swing")
+                if swing_act in (SWING_OFF, SWING_VERTICAL):
+                    await self._device.command("hor_swing")
+
+            if swing_mode == SWING_VERTICAL and swing_act != SWING_VERTICAL:
+                if swing_act in (SWING_OFF, SWING_HORIZONTAL):
+                    await self._device.command("vert_swing")
+                if swing_act in (SWING_BOTH, SWING_HORIZONTAL):
+                    await self._device.command("hor_dir")
+
+            if swing_mode == SWING_HORIZONTAL and swing_act != SWING_HORIZONTAL:
+                if swing_act in (SWING_BOTH, SWING_VERTICAL):
+                    await self._device.command("vert_dir")
+                if swing_act in (SWING_OFF, SWING_VERTICAL):
+                    await self._device.command("hor_swing")
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set new preset mode."""
+        if self._on == "1":
```

Comment:
Should we invert the check and log a message why we can't perform the action and return, instead?

Reply:
If you think it is better there is no problem. My idea was trivially that, if the machine is off, don't expect it to respond to commands.

Reply:
Alternatively, selecting a preset could turn on the machine. For the other commands (set temp, swing and fan modes) I could log the machine off warning.

Reply:
We should make it easy for the user to do what the user wants to do. If the device needs a turn on call before the preset selection, it's probably best to call turn on automatically if needed when the user sets a preset.

Should we do the same for the other commands?

Reply:
It makes sense that the user expects the presets to turn on the air conditioner, but the temperature, fan speed and swing parameters have no meaning except in relation to the current operating mode of the air conditioner.
But even for the presets there would be a wider discussion: the ignition will take place in the last mode set on the air conditioner, unknown to us, which may not be what the user wants.

Reply:
Ok. You should take this decision, as you know the device best. I can only give high level advice here. :+1:

Reply:
Is it okay to log in at the warning level in the cases indicated above?

Reply:
That's good, I think.